teku/src/main/java/tech/pegasys/teku/services/BeaconNodeServiceController.java

```diff
@@ -29,27 +32,44 @@ public BeaconNodeServiceController(
       TekuConfiguration tekuConfig, final ServiceConfig serviceConfig) {
     // Note services will be started in the order they are added here.
     services.add(new StorageService(serviceConfig, tekuConfig.storageConfiguration()));
+    Optional<ExecutionClientProvider> maybeExecutionClientProvider = Optional.empty();
     if (tekuConfig.executionEngine().isEnabled()) {
       // Need to make sure the execution engine is listening before starting the beacon chain
-      services.add(new ExecutionEngineService(serviceConfig, tekuConfig.executionEngine()));
+      final ExecutionClientProvider executionClientProvider =
+          ExecutionClientProvider.create(
+              tekuConfig.executionEngine().getEndpoint(),
+              serviceConfig.getTimeProvider(),
+              Optional.of(EXECUTION_TIMEOUT),
+              tekuConfig.executionEngine().getJwtSecretFile(),
+              serviceConfig.getDataDirLayout().getBeaconDataDirectory());
+      services.add(
+          new ExecutionEngineService(
+              serviceConfig, tekuConfig.executionEngine(), executionClientProvider));
+      if (!executionClientProvider.isStub()) {
+        maybeExecutionClientProvider = Optional.of(executionClientProvider);
+      }
```

Comment:
We shouldn't have this kind of creation logic at this level - the `ExecutionEngineService` needs to be responsible for creating everything it needs.  Otherwise we wind up with a huge mess of initialisation code for everything up at this level.

It's a bit unfortunate but probably reasonable to have a `Optional<ExecutionClientProvider> getExecutionClientProvider()` method on `ExecutionEngineService` so we can get the provider out and use it in pow chain service - breaks the service encapsulation a bit but not a lot of other options.